### ISSUE TITLE
Uri support for "/" syntax 

### ISSUE THREAD

**Issue by kamilkloch** [softwaremill/sttp#210]
What do you think about adding support for the following syntax (like in https://http4s.org/v0.20/uri/)?
```scala
val root = uri"/srv"
val service = root / "service"
```

**Comment by adamw** [comment]
There are two parts here:

1. `val root = uri"/srv"`

what would be the scheme/host/port in such an uri?

2.  `val service = anExistingUri / "service"`

that would be quite easy to implement, extending the path part. However the primary method of building an uri in sttp is through string interpolation; what's your use-case for this method? (maybe it's already covered)

**Comment by kamilkloch** [comment]
I am implementing endpoints for the client (have to do manually until tapir rocks on JS ;)). 
All endpoints share the same `root` uri. In the individual methods I would like to use the pattern
```scala
sttp.get(root / "serviceX")
```
What would be your take?

**Comment by adamw** [comment]
Would `uri"$root/serviceX"` work for you?

As for relative uris - these are not supported currently, the host is always required. I suppose in the scala.js context this would make sense, as you write. We'd have to make host an `Option` then.

**Comment by kamilkloch** [comment]
> Would `uri"$root/serviceX"` work for you?
This my current workaround, but I am not comfortable with it. I think it is broken:
```scala
val a = uri"http://google.com?a=b"
println(uri"$a/search")
```
```
http://google.com?a=b/search
``` 

> As for relative uris - these are not supported currently, the host is always required. I suppose in the scala.js context this would make sense, as you write. We'd have to make host an `Option` then.

I am happy to submit a PR, if that makes sense.

**Comment by adamw** [comment]
Ah ... well as a first step, I would try to fix `uri"$a/search"` so that it properly treats query parameters/paths with trailing slashes.

As for relative URIs, the problem would be changing the `Uri` class to have optional fields - as that would be a major breaking change. Probably would require a major release, so I would hold that for now.

PRs are of course always welcome :)

**Comment by kamilkloch** [comment]
@adamw please take a look at the trailing slashes fix. I could not find a more elegant way.

**Comment by adamw** [comment]
Released in 1.6.6, thanks!
